src/sagemaker/jumpstart/types.py

```diff
@@ -311,6 +312,73 @@ def to_json(self) -> Dict[str, Any]:
         return json_obj
 
 
+class JumpStartInstanceTypeVariants(JumpStartDataHolderType):
+    """Data class for JumpStart instance type variants."""
+
+    __slots__ = [
+        "regional_aliases",
+        "variants",
+    ]
+
+    def __init__(self, spec: Optional[Dict[str, Any]]):
+        """Initializes a JumpStartInstanceTypeVariants object from its json representation.
+
+        Args:
+            spec (Dict[str, Any]): Dictionary representation of instance type variants.
+        """
+        self.from_json(spec)
+
+    def from_json(self, json_obj: Optional[Dict[str, Any]]) -> None:
+        """Sets fields in object based on json.
+
+        Args:
+            json_obj (Dict[str, Any]): Dictionary representation of instance type variants.
+        """
+
+        if json_obj is None:
+            return
+
+        self.regional_aliases: dict = json_obj["regional_aliases"]
+        self.variants: dict = json_obj["variants"]
+
+    def to_json(self) -> Dict[str, Any]:
+        """Returns json representation of JumpStartInstanceTypeVariants object."""
+        json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
+        return json_obj
+
+    def get_image_uri(self, instance_type: str, region: str) -> Optional[str]:
+        """Returns image uri from instance type and region.
+
+        Returns None if no instance type is available or found.
+        """
+
+        image_uri_alias: Optional[str] = None
+        if instance_type in self.variants:
+            image_uri_alias = self.variants[instance_type]["regional_properties"].get("image_uri")
+        else:
+            instance_type_family = get_instance_type_family(instance_type)
+
+            if instance_type_family in {"", None}:
+                return None
+
+            image_uri_alias = (
+                self.variants[instance_type_family]["regional_properties"].get("image_uri")
+                if instance_type_family in self.variants
+                else None
+            )
+
+        if image_uri_alias is None:
+            return image_uri_alias
+
+        if not image_uri_alias.startswith("$"):
+            raise TypeError("All image uris should map to an alias and start with '$'.")
+
+        if region not in self.regional_aliases:
+            return None
+        alias_value = self.regional_aliases[region][image_uri_alias[1:]]
+        return alias_value
```

Comment:
question: if the instance type doesn't match a variant, how about:
- check if it's GPU instance family, if yes, map to `p3` variant if it exists
- otherwise, map to `m5` variant if it exists?

Reply:
i'd rather not, that would be embedding specific logic in the SDK, we want to keep this as generic as possible